pkg/cmd/stack/deploy.go

```diff
@@ -150,13 +160,156 @@ func deploy(ctx context.Context, s *model.Stack, wait bool, c *kubernetes.Client
+	spinner.Start()
+	return nil
+}
+
+func isVolumeDeployed(volumeName string, deployedVolumes []string) bool {
+	for _, deployedVolumeName := range deployedVolumes {
+		if deployedVolumeName == volumeName {
+			return true
+		}
+	}
+	return false
+}
+
+func isSvcDeployed(svcName string, deployedSvcs map[string]bool) bool {
+	return deployedSvcs[svcName]
+}
+
+func canSvcBeDeployed(ctx context.Context, stack *model.Stack, svcName string, client kubernetes.Interface, config *rest.Config) bool {
+	for dependentSvc, condition := range stack.Services[svcName].DependsOn {
+		if !isSvcReady(ctx, stack, dependentSvc, condition, client, config) {
+			return false
+		}
+	}
+	return true
+}
+
+func getDependingFailedJobs(ctx context.Context, stack *model.Stack, svcName string, client kubernetes.Interface, config *rest.Config) []string {
+	svc := stack.Services[svcName]
+	dependingJobs := make([]string, 0)
+	for dependingSvc := range svc.DependsOn {
+		if stack.Services[dependingSvc].RestartPolicy != apiv1.RestartPolicyAlways {
+			dependingJobs = append(dependingJobs, dependingSvc)
+		}
+	}
+	failedJobs := make([]string, 0)
+	for _, jobName := range dependingJobs {
+		if jobs.IsFailed(ctx, stack.Namespace, jobName, client) {
+			failedJobs = append(failedJobs, jobName)
+		}
+	}
+	return failedJobs
+}
+
+func isSvcReady(ctx context.Context, stack *model.Stack, dependentSvcName string, condition model.DependsOnConditionSpec, client kubernetes.Interface, config *rest.Config) bool {
+	svc := stack.Services[dependentSvcName]
+
+	switch condition.Condition {
+	case model.DependsOnServiceRunning:
+		return isSvcRunning(ctx, svc, stack.Namespace, dependentSvcName, client)
+	case model.DependsOnServiceHealthy:
+		return isSvcHealthy(ctx, svc, stack, dependentSvcName, client, config)
+	case model.DependsOnServiceCompleted:
+		if jobs.IsSuccedded(ctx, stack.Namespace, dependentSvcName, client) {
+			return true
+		}
+	}
+	return false
+}
+
+func getPodName(ctx context.Context, stack *model.Stack, svcName string, client kubernetes.Interface) string {
+	svcLabels := map[string]string{model.StackNameLabel: stack.Name, model.StackServiceNameLabel: svcName}
+
+	p, err := pods.GetBySelector(ctx, stack.Namespace, svcLabels, client)
+	if err != nil {
+		return ""
+	}
+	return p.Name
+}
+
+func isSvcRunning(ctx context.Context, svc *model.Service, namespace, svcName string, client kubernetes.Interface) bool {
+
+	switch {
+	case svc.IsDeployment():
+		if deployments.IsRunning(ctx, namespace, svcName, client) {
+			return true
+		}
+	case svc.IsStatefulset():
+		if statefulsets.IsRunning(ctx, namespace, svcName, client) {
+			return true
+		}
+	case svc.IsJob():
+		if jobs.IsRunning(ctx, namespace, svcName, client) {
+			return true
+		}
+	}
+	return false
+}
+
+func isSvcHealthy(ctx context.Context, svc *model.Service, stack *model.Stack, svcName string, client kubernetes.Interface, config *rest.Config) bool {
+	if isSvcRunning(ctx, svc, stack.Namespace, svcName, client) {
```

Comment:
```suggestion
	if !isSvcRunning(ctx, svc, stack.Namespace, svcName, client) {
	    return false
	}
```
This way the overal indentation is reduced